packages/contract/src/test-utils/test-node-launcher.test.ts

```diff
@@ -0,0 +1,254 @@
+import { Provider, sleepUntilTrue } from '@fuel-ts/account';
+import { WalletConfig } from '@fuel-ts/account/test-utils';
+import * as launchCustomProviderAndGetWalletsMod from '@fuel-ts/account/test-utils';
+import { FuelError } from '@fuel-ts/errors';
+import { expectToThrowFuelError, safeExec } from '@fuel-ts/errors/test-utils';
+import type { ChainConfig } from '@fuel-ts/utils';
+import { randomInt, randomUUID } from 'crypto';
+import { existsSync, mkdirSync, readFileSync, rmSync } from 'fs';
+import { writeFile } from 'fs/promises';
+import http from 'http';
+import os from 'os';
+import { join } from 'path';
+import { parse } from 'url';
+
+import { TestNodeLauncher } from './test-node-launcher';
+
+const pathToContractRootDir = join(__dirname, '../../test/fixtures/simple-contract');
+
+async function generateChainConfigFile(chainName: string): Promise<[string, () => void]> {
+  const chainConfig = JSON.parse(
+    readFileSync(
+      join(__dirname, '../../../../', '.fuel-core', 'configs', 'chainConfig.json'),
+      'utf-8'
+    )
+  ) as ChainConfig;
+
+  chainConfig.chain_name = chainName;
+
+  const tempDirPath = join(os.tmpdir(), '.fuels-ts', randomUUID());
+
+  if (!existsSync(tempDirPath)) {
+    mkdirSync(tempDirPath, { recursive: true });
+  }
+  const chainConfigPath = join(tempDirPath, '.chainConfig.json');
+
+  // Write a temporary chain configuration file.
+  await writeFile(chainConfigPath, JSON.stringify(chainConfig), 'utf-8');
+
+  return [chainConfigPath, () => rmSync(tempDirPath, { recursive: true, force: true })];
+}
+
+async function urlIsLive(url: string): Promise<boolean> {
+  return new Promise((resolve) => {
+    const req = http.request(parse(url), () => {
+      resolve(true);
+    });
+
+    req.on('error', () => resolve(false));
+
+    req.end();
+  });
+}
```

Comment:
Duplicated with `packages/account/src/test-utils/utils.ts`?

Reply:
Yes, I moved it all into `utils/test-utils`: https://github.com/FuelLabs/fuels-ts/pull/1356/commits/ecedbf9106c83bb8c12de2503090fc80d61ab6d0